pypapi/papi_high.py

```diff
@@ -1,296 +1,108 @@
     """
-    cvalues = ffi.new("long long[]", values)
-    rcode = lib.PAPI_accum_counters(cvalues, len(values))
-    return rcode, ffi.unpack(cvalues, len(values))
-
+    cregion = ffi.new("char[]", region.encode("ascii"))
+    rcode = lib.PAPI_hl_region_begin(cregion)
+    return rcode
 
-# int PAPI_num_counters(void);
-def num_counters():
-    """Get the number of hardware counters available on the system.
-
-    :rtype: int
 
-    :raises PapiInvalidValueError: ``papi.h`` is different from the version
-        used to compile the PAPI library.
-    :raises PapiNoMemoryError: Insufficient memory to complete the operation.
-    :raises PapiSystemError: A system or C library call failed inside PAPI.
-    """
-    return lib.PAPI_num_counters()
+# int PAPI_hl_read(const char* region);
+def hl_read(region):
+    """hl_read(region):
 
+    Read performance events inside of a region and store the difference to the corresponding beginning of the region
 
-# int PAPI_num_components(void);
-def num_components():
-    """Get the number of components available on the system.
+    :param (string) region: name of instrumented region
 
+    :returns: Operation status
     :rtype: int
-    """
-    return lib.PAPI_num_components()
-
-
-# int PAPI_read_counters(long long * values, int array_len);
-@papi_error
-def read_counters():
-    """read_counters()
-
-    Get current counts and reset counters.
-
-    :rtype: list(int)
-
-    :raises PapiInvalidValueError: One or more of the arguments is invalid
-        (this error should not happen with PyPAPI).
-    :raises PapiSystemError: A system or C library call failed inside PAPI.
-    """
-    values = ffi.new("long long[]", _counter_count)
-    rcode = lib.PAPI_read_counters(values, _counter_count)
-    return rcode, ffi.unpack(values, _counter_count)
-
-
-# int PAPI_start_counters(int *events, int array_len);
-@papi_error
-def start_counters(events):
-    """start_counters(events)
-
-    Start counting hardware events.
-
-    :param list events: a list of events to count (from :doc:`events`)
 
     :raises PapiInvalidValueError: One or more of the arguments is invalid.
-    :raises PapiIsRunningError: Counters have already been started, you must
-        call :py:func:`stop_counters` before you call this function again.
     :raises PapiSystemError: A system or C library call failed inside PAPI.
-    :raises PapiNoMemoryError: Insufficient memory to complete the operation.
-    :raises PapiConflictError: The underlying counter hardware cannot count
-        this event and other events in the EventSet simultaneously.
-    :raises PapiNoEventError: The PAPI preset is not available on the
-        underlying hardware.
     """
-    global _counter_count
-    _counter_count = len(events)
-
-    events_ = ffi.new("int[]", events)
-    array_len = len(events)
-
-    rcode = lib.PAPI_start_counters(events_, array_len)
+    cregion = ffi.new("char[]", region.encode("ascii"))
+    rcode = lib.PAPI_hl_region(cregion)
```

Comment:
Getting `AttributeError: cffi library 'pypapi._papi' has no function, constant or global variable named 'PAPI_hl_region'. Did you mean: 'PAPI_hl_read'?` when trying to call this.

Reply:
@WaDelma thanks for pointing out these problems. 
I fixed them on the commits below.
Could you test the lib?

I still have to validate if the changes from NumPy to CTypes that I made did not mess any masking operations.